### PR TITLE
feat: View patient's contact command

### LINKED ISSUE #26: As a user, I can view a patient's profile
so that I can contact the patient for follow up appointment.

The tasks need to complete this user story include:
- [ ] Update the Ui
- [ ] Add view contact information feature

### PR DESCRIPTION
**Pull request by clement0010**
This PR extends a new feature to view patient contacts in Medbook 📕 and closes #26 

## New ✨
- Medbook now support the viewing of patient's contacts along with their basic details.
- Medbook UI now displays the patient's contact list.

## Improvement 🎈
- Update JSON test cases.
- Update storage to save `Contacts`.

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 class JsonAdaptedPerson {
 
     public static final String MISSING_FIELD_MESSAGE_FORMAT = "Person's %s field is missing!";
+    public static final String MESSAGE_DUPLICATE_CONTACT = "Contact list contains duplicate contact(s).";
```
Comment (sibinhho99-nus) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:27]: Unused variable

**Review by sibinhho99-nus — COMMENTED** [review]
Commit messages should follow https://se-education.org/guides/conventions/git.html

**Review by sibinhho99-nus — APPROVED** [review]

**Review by sibinhho99-nus — APPROVED** [review]
Looks good, apart from some minor issues

File: src/main/java/seedu/address/logic/commands/contact/AddContactCommand.java [hunk 2]
```diff
@@ -39,43 +39,35 @@
     public static final String MESSAGE_MISSING_PATIENT = "This patient does not exists in Medbook";
 
     // Identifier
-    private final Nric nric;
+    private final Nric ownerNric;
 
     private final Contact toAdd;
 
     /**
      * Creates an AddContactCommand to add the specified {@code Patient}
```
Comment (sibinhho99-nus) [inline src/main/java/seedu/address/logic/commands/contact/AddContactCommand.java:47]: Add the specified Contact to owner?

File: src/main/java/seedu/address/logic/commands/contact/ViewContactCommand.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.CommandType;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.contact.ContactWithNricPredicate;
+import seedu.address.model.person.Nric;
+import seedu.address.model.person.NricPredicate;
+
+/**
+ * Lists all persons in the address book to the user.
```
Comment (sibinhho99-nus) [inline src/main/java/seedu/address/logic/commands/contact/ViewContactCommand.java:17]: Lists all contacts associated with a certain owner to the user?

File: src/main/java/seedu/address/logic/commands/contact/AddContactCommand.java (hunk 2 above)
Comment (clement0010) [reply src/main/java/seedu/address/logic/commands/contact/AddContactCommand.java:47]: Could u elaborate

File: src/main/java/seedu/address/logic/commands/contact/ViewContactCommand.java (hunk 3 above)
Comment (clement0010) [reply src/main/java/seedu/address/logic/commands/contact/ViewContactCommand.java:17]: Good catch. Perhaps u could use the [suggestion](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request) feature in code review next time to highlight the change.

File: src/main/java/seedu/address/logic/commands/contact/AddContactCommand.java (hunk 2 above)
Comment (sibinhho99-nus) [reply src/main/java/seedu/address/logic/commands/contact/AddContactCommand.java:47]: I thought the AddContactCommand is used to add a contact to a certain Patient/ Person (identified by the parameter `ownerNric`). I feel this description is a bit ambiguous in describing what AddContactCommand does

**Pull request merged**